packages/bygger/server/backend/migration/migrationScripts.js

```diff
@@ -40,12 +40,42 @@ function getEditScript(editOptions, logger = []) {
   };
 }
 
+function hasChangesToPropertiesWhichCanBreakDependencies(diff) {
+  return (
+    diff.key_NEW || // Keys are used to look up submissions for components, and is the most common dependency
+    diff.values_NEW || // Changes to values for Radiopanel or Flervalg components can break references that are depending on a specific value
+    diff.data_NEW ||
+    (diff.data && diff.data.values_NEW) // Values for Nedtrekksliste are stored in data.values, so changes to data or data.values can be breaking
+  );
+}
+
 async function fetchForms(url) {
   return await fetchWithErrorHandling(url, {
     method: "GET",
   });
 }
 
+function getBreakingChanges(form, changes) {
+  return changes
+    .filter((affected) => affected.diff)
+    .map((affected) => affected.diff)
+    .filter((diff) => hasChangesToPropertiesWhichCanBreakDependencies(diff))
+    .reduce((diffsWithBreakingChanges, currentDiff) => {
```

Comment:
er det mulig å bruke map eller flatMap i stedet for reduce?

Reply:
Erstattet med flatMap